### DISCUSSION TITLE
Why need both BC Pressure and Dirichlet in pore pressure in Porous Flow Tutorial 4

### DISCUSSION THREAD

**Discussion by alexguev** [idaholab/moose#20977] [Q&A Modules: General] [answered]
Hello,

As I'm interested in injecting a fluid into a domain from a boundary, similarly in that regard to the tutorial 4 of the Porous Flow module, I was wondering why there were both the following BC:
```
  [constant_injection_porepressure]
    type = DirichletBC
    variable = porepressure
    value = 1E6
    boundary = injection_area
  []
```
and
```
  [cavity_pressure_x]
    type = Pressure
    boundary = injection_area
    variable = disp_x
    component = 0
    factor = 1E6
    use_displaced_mesh = false
  []
  [cavity_pressure_y]
    type = Pressure
    boundary = injection_area
    variable = disp_y
    component = 1
    factor = 1E6
    use_displaced_mesh = false
  []
```
From playing with both BC, I can see why they are both useful - I'd struggle to get the inner boundary (where the injection is applied) to expand without the Pressure BC. How to justify though, from a more theoretical point of view, that we need the Pressure BC on top of the Dirichlet BC?
I understand at first sight that for the Dirichlet BC, the injection pressure affects the solid through the momentum balance, whereas for the Pressure BC, the pressure is a traction, so only affecting the stress times the normal to the boundary. But on paper, in 1D for instance, it seems to (almost) has the same effect, unless mistaken.

 I feel @WilkAndy might know :) 

Cheers

**Comment by GiudGiud** [comment]
@WilkAndy do you know?

**Comment by WilkAndy** [comment, marked as answer]
I feel like you know the answer @alexguev , but you don't know that you know!  You can write out the equations and find the answer, but here's my wordy explanation....

The `DirichletBC` is needed because the borehole is at 1MPa pressure.  That fluid is also "pushing" on the rock skeleton: the fluid is mechanically loading the rock.  Eg, imagine a totally non-porous rock: it's pretty clear the borehole pressure will "push" on the rock.  Mathematically this means the porepressure is acting as an applied total stress (not effective stress) at the borehole boundary.   There is some discussion in the "Barometric and oceanic loading" section of https://mooseframework.inl.gov/modules/porous_flow/tidal.html .   All stresses in PorousFlow are effective stresses (eg, for the constitutive relations, and the output you see in exodus files) *except* for any Neumann boundary conditions, which are written in terms of total stresses - see discussion around Eqn(12) of https://mooseframework.inl.gov/modules/porous_flow/governing_equations.html .  Hence, we need two `Pressure` BCs for the total stress (the two mean the applied pressure is acting in the normal direction).

**Reply by alexguev** [reply]
Makes perfect sense, thanks @WilkAndy! I mistakenly thought the ``Pressure`` BC was acting on the effective stress.

**Reply by alexguev** [reply]
Follow-up question if I may @WilkAndy : how should I proceed to impose constant flow rate instead of constant pressure? Instead of ``DirichletBC`` for the pore pressure I would use ``NeumannBC``, but what about the ``Pressure`` BC for the total stress?

**Reply by WilkAndy** [reply]
OK, that is more complicated, but still quite possible.  Have a look at https://mooseframework.inl.gov/modules/porous_flow/thm_example.html .  This isn't exactly your problem, so don't copy it exactly (it uses specific and rather hypothetical assumptions in order to compare with a case that has an analytical solution) but it explains the use of a postprocessor to record fluid pressure, and then apply that to the total stress.  If you understand the opaque comment `note, this lags` in the input file, you'll have understood everything!

**Reply by alexguev** [reply]
That works, thanks, provided one is careful to have timesteps small enough. I guess a potential issue might be when the pressure at the injection boundary is not homogeneous, like when the porosity is prescribed through a spatial field for instance; but then, there might exist some way to assign the pressure at each node - that will be for another day (unless there is an easy solution).
